src/js/hooks/cycleCount/useResolveStep.js

```diff
@@ -103,25 +104,50 @@ const useResolveStep = () => {
           dateCounted: null,
           countedBy: null,
           rootCause: mapRootCauseToSelectedOption(item?.discrepancyReasonCode),
-        };
+        }];
       }
 
+      const groupedByCountIndex = _.groupBy(itemsToMerge, 'countIndex');
       const maxCountIndex = _.maxBy(itemsToMerge, 'countIndex').countIndex;
       const itemFromCount = _.find(itemsToMerge, (item) => item.countIndex === maxCountIndex - 1);
-      const itemFromResolve = _.find(itemsToMerge, (item) => item.countIndex === maxCountIndex);
-      return {
-        ...itemFromCount,
-        ...itemFromResolve,
-        commentFromCount: itemFromCount?.comment,
-        quantityRecounted: itemFromResolve?.quantityCounted,
-        quantityCounted: itemFromCount?.quantityCounted,
-        quantityVariance: itemFromCount?.quantityVariance,
-        dateCounted: itemFromCount?.dateCounted,
-        dateRecounted: itemFromResolve?.dateCounted,
-        countedBy: itemFromCount?.assignee,
-        recountedBy: itemFromResolve?.assignee,
-        rootCause: mapRootCauseToSelectedOption(itemFromResolve?.discrepancyReasonCode),
-      };
+      const itemsFromResolve = groupedByCountIndex[maxCountIndex] || [];
+
+      return itemsFromResolve.map((item, index) => {
+        // If this is the first item with maxCountIndex and itemFromCount exists,
+        // merge data from both cycles. We do this because we wanted to combine data from
+        // countIndex 0 with the first countIndex 1 to avoid duplicating info from both counts
+        if (index === 0 && itemFromCount) {
+          return {
+            ...itemFromCount,
+            ...item,
+            commentFromCount: itemFromCount?.comment,
+            quantityRecounted: item?.quantityCounted,
+            quantityCounted: itemFromCount?.quantityCounted,
+            quantityVariance: itemFromCount?.quantityVariance,
+            dateCounted: itemFromCount?.dateCounted,
+            dateRecounted: item?.dateCounted,
+            countedBy: itemFromCount?.assignee,
+            recountedBy: item?.assignee,
+            rootCause: mapRootCauseToSelectedOption(item?.discrepancyReasonCode),
+          };
+        }
+
+        // For example we have countIndex 0, 1, 1: the item with
+        // countIndex 0 merges with the first item with countIndex 1,
+        // while the next item with countIndex 1 remains a separate record without merging
```

Comment:
is this really the desired behaviour? Why don't we want to display the rows as identical? What happens if the user deletes the other one (the one with more information)?

I think we should confirm the desired behaviour with Manon.